### PR TITLE
Using crd to store vhost relationship with deployment

### PR DESCRIPTION
**Pull request by pranjaljain**
Till now we had an implementation to store relationship between virtualhost instance and deployment in virtualHostCloudProviders, changing it to store via apiserver in CRDs.
As we are planning this feature to be released this sprint, there are no active vhosts instances in the landscapes, so no migration is needed.

### CODE REVIEW

**Comment by pranjaljain** [comment]
@kamath-prasad Yes there are bosh release changes as well. https://github.com/cloudfoundry-incubator/service-fabrik-boshrelease/pull/135
Apologies for bringing this on last day of sprint.

**Comment by kamath-prasad** [comment]
@pranjaljain  : Also pls fix the tests , the validator fails with 

```
$ npm run -s test-coverall
========================= BROKER ==========================
circuit breaker is disabled
The command "npm run -s test-coverall" exited with 1.
```

File: broker/config/crds/deployment.servicefabrik.io_v1alpha1_virtualhosts.yaml [hunk 1]
```diff
@@ -25,6 +25,8 @@ spec:
             options:
               type: string
           type: object
+        operatorMetadata:
```
Comment (subhankarc) [inline broker/config/crds/deployment.servicefabrik.io_v1alpha1_virtualhosts.yaml]: naming conventions are generally all small cases.

Comment (subhankarc) [reply broker/config/crds/deployment.servicefabrik.io_v1alpha1_virtualhosts.yaml]: should be fine..https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md

File: operators/virtualhost-operator/VirtualHostRelationMapper.js [hunk 2]
```diff
@@ -36,23 +35,20 @@ class VirtualHostRelationMapper {
...
   }
 
   deleteVirtualHostRelation(instanceId) {
     logger.info(`Deleting deployment name relation for virtual_host instance id : '${instanceId}'`);
-    const options = {
-      'instance_guid': instanceId
-    };
-    return this.virtualHostStore.removeFile(options)
-      .then(() => this.deleteCacheEntry(instanceId));
+    return this.deleteCacheEntry(instanceId);
```
[the remark this answers is not in the document]
Comment (pranjaljain) [reply operators/virtualhost-operator/VirtualHostRelationMapper.js]: delete resource is done at VirtualHostOperator https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/471/files#diff-f54e6ac49e4398933cc0febd9c8c562fR82

**Comment by pranjaljain** [comment]
@kamath-prasad Can the PR be merged now? 
Uncovered lines in broker/lib/fabrik/VirtualHostRelationMapper.js are because not all the code present in lib/fabrik code is used. But I had to duplicated the changes I did in the VirtualHostOperator to this file also so that dashboard info call work.

**Pull request merged**